### ISSUE TITLE
Installation fails in zsh

### ISSUE THREAD

**Issue by ghost** [markeissler/gvm2#3]
```
$ curl -sSL https://raw.githubusercontent.com/markeissler/gvm2/master/binscripts/gvm-installer | zsh
Cloning from https://github.com/markeissler/gvm2.git to /Users/wes/.gvm
zsh: command not found: git clone --quiet
ERROR: Failed to clone from https://github.com/markeissler/gvm2.git into /Users/wes/.gvm
```

I have git installed, so that's not the problem.

**Comment by markeissler** [comment]
Do you have `git` installed? At this time, __GVM2__ (like the legacy __GVM__ project) is installed by cloning the __GVM2__ source from GitHub. This install method is likely to change all together before version 1.

**Comment by ghost** [comment]
As I wrote in my original comment, yes I have git installed. I think this might have to do with how the command is being stored/invoked

**Comment by markeissler** [comment]
Ok. Found the problem.

But then found a different problem. I don't use `zsh` but I'll try to get this working again with that shell for the next release.

Thanks for reporting this.

**Comment by krostar** [comment]
I have the same problem:

```sh
$> git --version
git version 2.11.0
$> zsh --version
zsh 5.2 (x86_64-ubuntu-linux-gnu) 
$> curl -sSL https://raw.githubusercontent.com/markeissler/gvm2/master/binscripts/gvm-installer | zsh
Cloning from https://github.com/markeissler/gvm2.git to /home/exahlkiq/.gvm
zsh: command not found: git clone --quiet
zsh: command not found: __gvm_display_error
cd: no such file or directory: /home/exahlkiq/.gvm
zsh: command not found: __gvm_display_error
mv: cannot stat '/home/exahlkiq/.gvm/.git': No such file or directory
zsh: no such file or directory: /home/exahlkiq/.gvm/scripts/gvm
zsh: no such file or directory: /home/exahlkiq/.gvm/scripts/gvm
Created profile for existing install of Go at 
check_existing_go:20: command not found: __gvm_display_error
check_existing_go:21: command not found: __gvm_display_error
check_existing_go:22: command not found: __gvm_display_error
check_existing_go:23: no such file or directory: /home/exahlkiq/.gvm/environments/system
check_existing_go:36: command not found: cp
check_existing_go:39: no such file or directory: /home/exahlkiq/.gvm/environments/system@global
Installed GVM v

Please restart your terminal session or to get started right away run
 `source /home/exahlkiq/.gvm/scripts/gvm`

Support for auto selection of Go version and GVM2 pkgset (via .go-version
and .go-pkgset files) requires that GVM2 is loaded after RVM (if installed).
For bash, you may need to manually update your .profile or .bash_profile
files; for zsh, you may need to manually update your .zlogin file.

See the README.md file for more information.
   
```

**Comment by krostar** [comment]
When I use `bash` instead of `zsh` on installation, I have an error when sourcing gvm:

```sh
$> curl -sSL https://raw.githubusercontent.com/markeissler/gvm2/master/binscripts/gvm-installer | bash
Cloning from https://github.com/markeissler/gvm2.git to /home/exahlkiq/.gvm
No existing Go versions detected
Installed GVM v0.10.4

Please restart your terminal session or to get started right away run
 `source /home/exahlkiq/.gvm/scripts/gvm`

Support for auto selection of Go version and GVM2 pkgset (via .go-version
and .go-pkgset files) requires that GVM2 is loaded after RVM (if installed).
For bash, you may need to manually update your .profile or .bash_profile
files; for zsh, you may need to manually update your .zlogin file.

See the README.md file for more information.

$> source /home/exahlkiq/.gvm/scripts/gvm
dep_load:source:12: no such file or directory: /home/exahlkiq/../function/_shell_compat.sh
dep_load:source:12: no such file or directory: /home/exahlkiq/../function/display_notices.sh
dep_load:source:12: no such file or directory: /home/exahlkiq/../function/locale_text.sh
dep_load:source:12: no such file or directory: /home/exahlkiq/pkgset_use.sh
dep_load:source:12: no such file or directory: /home/exahlkiq/use.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/_bash_pseudo_hash.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/_shell_compat.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/display_notices.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/find_installed.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/find_path_upwards.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/locale_text.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/munge_path.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/read_environment_file.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/resolve_fallback_version.sh
dep_load:source:17: no such file or directory: /home/exahlkiq/../function/resolve_fallback_pkgset.sh
/home/exahlkiq/.gvm/scripts/env/cd.sh:35: command not found: __gvm_is_function
/home/exahlkiq/.gvm/scripts/env/cd.sh:43: command not found: __gvm_is_function
/home/exahlkiq/.gvm/scripts/env/cd.sh:59: command not found: __gvm_munge_path
/home/exahlkiq/.gvm/scripts/env/cd.sh:60: command not found: __gvm_display_warning
cd:4: command not found: __gvm_is_function
cd:22: command not found: __gvm_read_environment_file
cd:31: command not found: __gvm_read_environment_file
cd:47: command not found: __gvm_resolve_fallback_version
cd:51: command not found: __gvm_resolve_fallback_pkgset
cd:55: command not found: setValueForKeyFakeAssocArray
cd:57: command not found: setValueForKeyFakeAssocArray
cd:70: command not found: valueForKeyFakeAssocArray
cd:73: command not found: valueForKeyFakeAssocArray
__gvmp_find_closest_dot_go_version:3: command not found: __gvm_find_path_upwards
```

**Comment by markeissler** [comment]
@krostar I'll take a look over the coming weekend.

It looks like the gvm path is not getting set properly.

**Comment by markeissler** [comment]
@krostar Quick update: This issue is being actively worked on. I may be releasing an bug fix to address the bash issues first as the zsh compatibility fixes might take a bit longer to complete and test. **So, if not 0.10.5 then 0.10.6.**

**Comment by krostar** [comment]
Dont worry, take your time, I'm using Golang with a classic installation and I'll try gvm2 again when the bug will be fixed.

Tell me if you need someone to test your bugfix

**Comment by markeissler** [comment]
@krostar I just released **GVM2 0.10.6** which unofficially supports zsh. It's tested on **zsh 5** and **zsh 5.3**. I say this is "unofficial" because my next task is to write all of the unit and integration tests for zsh.

You should do a clean install to test this release. Note that older versions of Go will no longer build on gcc6+ (on linux) which is similar to older versions no longer building on clang8+ (on macOS). The list of available Gos for building from source will be filtered to reflect this change in 0.11.0. You should always be able to install binaries with the `gvm install <goversion> --binary` command (that is is the correct order of arguments, it's the only remaining command I need to rewrite).

Let me know how it goes.

**Comment by krostar** [comment]
I'm currently really busy, I'll keep you in touch once I get some time to test this release (probably this week-end).
Anyway, thx for the release !

**Comment by krostar** [comment]
Hi !

I have the following errors after installing gvm2 and restarting zsh :
```sh
zle-line-init:3: bad output format specification
getColorCode:30: no matches found: 006[blue]
getColorCode:30: no matches found: 006[black]
getColorCode:30: no matches found: 006[black]
getColorCode:30: no matches found: 006[blue]
getColorCode:30: no matches found: 006[blue]
getColorCode:30: no matches found: 006[black]
getColorCode:30: no matches found: 006[black]
```
I found the GetColorCode function in the `.zprezto/modules/prompt/external/powerlevel9k/functions/colors.zsh` file.

I dont understand why and how gvm2 can mess with zprezto :/

**Comment by markeissler** [comment]
So you have [Prezto](https://github.com/sorin-ionescu/prezto) installed and then maybe also a theme?

**Comment by krostar** [comment]
Yep, I use prezto + powerlevel9k theme

**Comment by markeissler** [comment]
@krostar I have managed to replicate this problem and have also found the source cause.

It's going to take a bit of refactoring to fix this since its related to the fact that bash arrays start at index 0 and zsh arrays start at index 1. Fixes have been in place to address this problem in other areas but clearly this needs to be done in a more robust manner.

I'm going to target **0.10.8** for fixing this.

**Comment by markeissler** [comment]
@krostar The original issue reported here related to installation failures on zsh. That bug was addressed in 0.10.4-0.10.5. I'm closing this issue at this time. Please continue discussion of the theme and prompt problems in the related Issue #4 (gvm2 messes with my prompt).
